cmd/installer/subcommands/installer/command.go

```diff
@@ -10,60 +10,114 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"time"
 
 	"github.com/DataDog/datadog-agent/cmd/installer/command"
+	"github.com/DataDog/datadog-agent/pkg/fleet/bootstraper"
 	"github.com/DataDog/datadog-agent/pkg/fleet/installer"
 	"github.com/DataDog/datadog-agent/pkg/fleet/telemetry"
 	"github.com/spf13/cobra"
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
 )
 
+const (
+	envRegistry     = "DD_INSTALLER_REGISTRY"
+	envRegistryAuth = "DD_INSTALLER_REGISTRY_AUTH"
+	envAPIKey       = "DD_API_KEY"
+	envSite         = "DD_SITE"
+)
+
 // Commands returns the installer subcommands.
 func Commands(_ *command.GlobalParams) []*cobra.Command {
-	return []*cobra.Command{installCommand(), removeCommand(), installExperimentCommand(), removeExperimentCommand(), promoteExperimentCommand(), garbageCollectCommand()}
+	return []*cobra.Command{bootstrapCommand(), installCommand(), removeCommand(), installExperimentCommand(), removeExperimentCommand(), promoteExperimentCommand(), garbageCollectCommand()}
 }
 
-type installerCmd struct {
-	installer.Installer
-	t    *telemetry.Telemetry
-	ctx  context.Context
-	span ddtrace.Span
+type cmd struct {
+	t            *telemetry.Telemetry
+	ctx          context.Context
+	span         ddtrace.Span
+	registry     string
+	registryAuth string
+	apiKey       string
+	site         string
+}
+
+func newCmd(operation string) *cmd {
+	span, ctx := newSpan(operation)
+	registry := os.Getenv(envRegistry)
+	registryAuth := os.Getenv(envRegistryAuth)
+	apiKey := os.Getenv(envAPIKey)
+	site := os.Getenv(envSite)
+	return &cmd{
+		t:            newTelemetry(),
+		ctx:          ctx,
+		span:         span,
+		registry:     registry,
+		registryAuth: registryAuth,
+		apiKey:       apiKey,
+		site:         site,
+	}
 }
 
-func (i *installerCmd) Stop(err error) {
-	i.span.Finish(tracer.WithError(err))
-	if i.t != nil {
-		err := i.t.Stop(context.Background())
+func (c *cmd) Stop(err error) {
+	c.span.Finish(tracer.WithError(err))
+	if c.t != nil {
+		err := c.t.Stop(context.Background())
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "failed to stop telemetry: %v\n", err)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 Thought:** ‏Not a blocker but it'd be nice to define methods for error logs & regular logs to avoid confusion